### PR TITLE
refactored test/cmd/newapp to use wrapper functions

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/cc @deads2k

@bparees @csrwng PTAL -- please review and ensure that the expectations (success, failure, with/without text) are correct to what you expected from the tests previously.

Changes other than simple addition of wrappers are commented below

### CODE REVIEW

File: test/cmd/newapp.sh [hunk 1]
```diff
@@ -6,127 +6,127 @@ set -o pipefail
...
-[ "$(oc new-app ruby-helloworld-sample -o yaml | grep ADMIN_PASSWORD)" ]
+os::cmd::expect_success 'oc create -f examples/sample-app/application-template-stibuild.json'
+os::cmd::expect_success 'oc get template ruby-helloworld-sample'
+os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'MYSQL_USER'
+os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'MYSQL_PASSWORD'
+os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'ADMIN_USERNAME'
+os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'ADMIN_PASSWORD'
 
 # check search
-[ "$(oc new-app --search mysql | grep -E "Tags:\s+5.5, 5.6, latest")" ]
```
Comment (stevekuznetsov) [inline test/cmd/newapp.sh]: `grep -E` is implied.

File: test/cmd/newapp.sh [hunk 2]
```diff
@@ -6,127 +6,127 @@ set -o pipefail
...
 # override component names
-[ "$(oc new-app mysql --name=db | grep db)" ]
-oc new-app https://github.com/openshift/ruby-hello-world -l app=ruby
-oc delete all -l app=ruby
+os::cmd::expect_success_and_text 'oc new-app mysql --name=db' 'db'
+os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world -l app=ruby'
+os::cmd::expect_success 'oc delete all -l app=ruby'
 
 # check new-build
-[ "$(oc new-build mysql -o yaml 2>&1 | grep -F 'you must specify at least one source repository URL')" ]
```
Comment (stevekuznetsov) [inline test/cmd/newapp.sh]: `grep -F` doesn't seem to have any meaning here

**Comment by bparees** [comment]
lgtm [merge]

**Comment by stevekuznetsov** [comment]
Flake in e2e:
```
[INFO] Waiting for command to finish: 'oc get rc/failing-dc-1 --template={{.metadata.annotations}} | grep openshift.io/deployment.phase:Failed'...
[ ERR] Gave up waiting for: 'oc get rc/failing-dc-1 --template={{.metadata.annotations}} | grep openshift.io/deployment.phase:Failed'

[FAIL] !!!!! Test Failed !!!!
```

**Comment by stevekuznetsov** [comment]
@bparees please re-tag

**Comment by deads2k** [comment]
> @bparees please re-tag

Link to flake issue.

**Comment by deads2k** [comment]
re[merge]

**Comment by stevekuznetsov** [comment]
flake issue: https://github.com/openshift/origin/issues/6077

**Pull request merged**